hare/hare.go

```diff
@@ -148,6 +152,10 @@ func (h *Hare) onTick(id types.LayerID) {
 		h.lastLayer = id
 	}
 	h.layerLock.Unlock()
+	if !h.isSynced() {
```

Comment:
REmove this, @gavraz is working on a PR that better handle this case (there are more issues regarding early messages)

Reply:
should we remove this or merge this code and then later revise according to @gavraz PR?

Reply:
hmm I wanted to ease on Gavrad's merge, but either way is fine

Reply:
I've removed this call and the api should remain the same